subprojects/core/src/main/java/org/gradle/process/internal/worker/child/SystemApplicationClassLoaderWorker.java

```diff
@@ -138,13 +144,45 @@ public ServiceRegistry getServiceRegistry() {
             if (connection != null) {
                 connection.stop();
             }
+            unrecoverableErrorHandler.close();
             messagingServices.close();
             loggingManager.stop();
         }
 
         return null;
     }
 
+    private class PrintUnrecoverableErrorToFileHandler implements Action<Throwable> {
+        private final File gradleUserHomeDir;
+        private PrintStream ps;
+
+        private PrintUnrecoverableErrorToFileHandler(File gradleUserHomeDir) {
+            this.gradleUserHomeDir = gradleUserHomeDir;
+        }
+
+        @Override
+        public void execute(Throwable throwable) {
+            if (ps == null) {
+                String fileName = "worker-error-" + new SimpleDateFormat("yyyyMMddHHmmss").format(new Date()) + ".txt";
+                try {
+                    ps = new PrintStream(new File(gradleUserHomeDir, fileName));
```

Comment:
I'd put this in the "workers" directory under gradleUserHome.  It would be really great if we could eventually correlate this in the build process and when a worker disappears, look for the worker error log, but this is a good first step - at least we can locate and see the error.